### ISSUE TITLE
"Forward security: No", but SSL Labs shows it is properly supported

### ISSUE THREAD

**Issue by FabioPedretti** [jannispinter/indicatetls#26]
On many sites (example: https://www.kernel.org/ ) IndicateTLS (0.1.2) reports "Forward security: No", however Qualys SSL Labs shows it is properly supported:

https://dev.ssllabs.com/ssltest/analyze.html?d=www.kernel.org&s=147.75.58.133

> Forward Secrecy | Yes (with most browsers)                    ROBUST

![forward-secrecy](https://user-images.githubusercontent.com/3390837/73258687-8b065700-41c6-11ea-9607-338c0ae5b802.png)

**Comment by jannispinter** [comment]
Oh wow, this should not happen! It should say "Forward security: Yes", because the cipher suite is ECDHE. IndicateTLS evaluates the "Key exchange" field, but "Key exchange:" shows N/A for you, this is actually a bug in Firefox's SecurityInfo API where securityInfo.keaGroupName is sometimes undefined.

We could evaluate the cipher suite (instead of the key exchange field) for TLS 1.2 to obtain information about forward security and for TLS 1.3 we could set it to always say 'Yes'.

**Comment by rugk** [comment]
> this is actually a bug in Firefox's SecurityInfo API where securityInfo.keaGroupName is sometimes undefined.

Can reproduce this on https://www.kernel.org/. So maybe this bug should be repoted to Mozilla? Do we have any idea what causes this?

**Comment by jannispinter** [comment]
It works for me on https://www.kernel.org, can you try hitting `Ctrl + F5`?

It would also be interesting to see what the _Security_ tab for the network request in the developer console says, because this is essentially the same data that is passed to the SecurityInfo API.
![grafik](https://user-images.githubusercontent.com/1999743/73269545-505be900-41dd-11ea-87c8-b818d5c6fb62.png)

**Comment by rugk** [comment]
Yeah, now I also cannot reproduce it anymore. Maybe only on a fresh installation?
Also when I clicked the link https://www.kernel.org/ for the first time (middle click to open in new tab), the "?" icon remained in there.

**Comment by FabioPedretti** [comment]
Even with CTRL + F5 I still see the "Forward security: No".
I am using Firefox 72.0.1 (Ubuntu package).
Here is the security tab:
![firefox](https://user-images.githubusercontent.com/3390837/73270376-d3ca0a00-41de-11ea-9692-472dff5fb445.png)

**Comment by jannispinter** [comment]
Thanks! I have just released version v0.1.3 with a workaround for this. I have also included the Italian translation from @FabioPedretti 

> Also when I clicked the link https://www.kernel.org/ for the first time (middle click to open in new tab), the "?" icon remained in there.

I have observed this a few times myself, this especially occurs now on very simple pages and is related to #23. This bug is triggered by a race condition (some stuff is happening asynchronously, the data is read before it was written by another function). Is your network blazingly fast?
